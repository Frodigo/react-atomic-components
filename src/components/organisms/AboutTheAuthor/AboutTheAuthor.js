import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/Text/Heading';
import Link from '../../atoms/Link';

import classes from './AboutTheAuthor.module.css';

const AboutTheAuthor = (props) => {
    const {
        heading,
        name,
        image,
        description,
        ctaUrl,
        ctaLabel,
        ...rest
    } = props;

    const shouldDisplayCta = ctaUrl ? (
        <Link variant="secondary" href={ctaUrl}>
            {ctaLabel}
        </Link>
    ) : null;

    const shouldDisplayImage = image ? (
        <figure
            className={classes.figureDesktop}
            style={{
                backgroundImage: `url("${
                    image
                }")`
            }}
        />
    ) : null;

    return (
        <div className={classes.aboutTheAuthor} {...rest}>
            <div className={classes.inner}>
                {shouldDisplayImage}
                <article className={classes.content}>
                    <Heading type="h6">{heading}</Heading>
                    <p className={classes.authorName}>{name}</p>
                    <figure
                        className={classes.figure}
                        style={{
                            backgroundImage: `url("${image}")`
                        }}
                    />
                    <p className={classes.description}>{description}</p>
                    {shouldDisplayCta}
                </article>
            </div>
        </div>
    );
};

AboutTheAuthor.propTypes = {
    ctaLabel: PropTypes.string,
    ctaUrl: PropTypes.string,
    description: PropTypes.string,
    heading: PropTypes.string,
    image: PropTypes.string,
    name: PropTypes.string.isRequired
};

AboutTheAuthor.defaultProps = {
    heading: 'About the author'
};

export default AboutTheAuthor;
