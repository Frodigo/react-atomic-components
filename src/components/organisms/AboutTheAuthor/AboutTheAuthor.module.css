.aboutTheAuthor {
    border-top: solid 2px var(--color-primary);
    padding: var(--spacing-xxl) var(--spacing-md);
    text-align: center;
}

.authorName {
    font-size: 18px;
    text-transform: uppercase;
}

.figure {
    background-color: #d8d8d8;
    border: solid 1px #979797;
    border-radius: 50%;
    display: inline-block;
    height: 270px;
    margin: 30px auto;
    width: 270px;
    overflow: hidden;
}

.description {
  margin-bottom: var(--spacing-xxl);
}

.figureDesktop {
    composes: figure;
    display: none;
}

@media (min-width: 768px) {
    .inner {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
    }
}


@media (min-width: 1300px) {
    .aboutTheAuthor {
        text-align: left;
    }

    .inner {
        align-items: center;
        display: flex;
        max-width: var(--desktop-container-width);
    }

    .figure {
        display: none;
    }

    .figureDesktop {
        display: block;
    }

    .content {
        flex: 1 1 60%;
        padding-left: var(--spacing-xxl);
    }
}