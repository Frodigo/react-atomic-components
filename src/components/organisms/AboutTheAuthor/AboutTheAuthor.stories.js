import React from 'react';

import { default as AboutTheAuthorComponent } from './AboutTheAuthor';

export default {
    title: 'Organisms',
    component: AboutTheAuthorComponent
};

const Template = (args) => <AboutTheAuthorComponent {...args} />;

export const AboutTheAuthor = Template.bind({});
AboutTheAuthor.args = {
    name: 'John Kowalski',
    image: 'https://via.placeholder.com/270x270',
    description:
        'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.',
    ctaUrl: '#',
    ctaLabel: 'Read more'
};
