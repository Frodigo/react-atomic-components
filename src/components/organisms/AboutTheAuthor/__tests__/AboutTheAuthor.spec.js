import React from 'react';
import { render } from '@testing-library/react';

import AboutTheAuthor from '../AboutTheAuthor';

const AboutTheAuthorProps = {
    name: 'Marcin Kwiatkowski',
    image: 'https://via.placeholder.com/270x270',
    description:
        'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.',
    ctaUrl: '#',
    ctaLabel: 'Read more'
};

describe('AboutThAuthor should be rendered', () => {
    test('AboutThAuthor renders correctly', () => {
        const { container } = render(
            <AboutTheAuthor {...AboutTheAuthorProps} />
        );

        expect(container).toMatchSnapshot();
    });
});
