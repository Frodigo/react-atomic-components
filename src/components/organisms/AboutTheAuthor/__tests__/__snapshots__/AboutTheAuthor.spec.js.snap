// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutThAuthor should be rendered AboutThAuthor renders correctly 1`] = `
<div>
  <div
    class="aboutTheAuthor"
  >
    <div
      class="inner"
    >
      <figure
        class="figureDesktop"
        style="background-image: url(https://via.placeholder.com/270x270);"
      />
      <article
        class="content"
      >
        <h6
          class=""
        >
          About the author
        </h6>
        <p
          class="authorName"
        >
          Marcin Kwiatkowski
        </p>
        <figure
          class="figure"
          style="background-image: url(https://via.placeholder.com/270x270);"
        />
        <p
          class="description"
        >
          Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.
        </p>
        <a
          class="secondary"
          href="#"
        >
          Read more
        </a>
      </article>
    </div>
  </div>
</div>
`;
